HelloWorld_ST_Sensors/user/gps/gps.hpp

```diff
@@ -0,0 +1,66 @@
+#ifndef USER_GPS_H_
+#define USER_GPS_H_
+
+#include "mbed.h"
+#include <string.h>
+
+namespace gps_ns
+{
+    class gps_c
+    {
+    private:
+        static constexpr uint32_t UART_BUF_LEN = 400;
+        RawSerial gps_uart;
+        uint32_t index;
+
+        uint8_t check_gprmc_gpgga_line(char *check_for_rmc_gga_in_this_str);
+        uint8_t extract_gpgga_or_gprmc_line(char *gps_buff);
+
+        void gps_interrupt_cb(void);
+        void gps_struct_init(void);
+        void (*gps_process_fn_ptr)(const char *newline);
+
+    public:
+        gps_c(PinName tx, PinName rx, uint32_t baud) : gps_uart(tx, rx)
+        {
+            gps_uart.baud(baud);
+            gps_uart.attach(callback(this, &gps_c::gps_interrupt_cb));
+        }
+
+        typedef struct gps_data
+        {
+            char latitude[15];
```

Comment:
latittude, longitude will be double values
lat dir and long dir can be enum values
validity can be a bool value

You can consider making different structs for GGA and RMC